--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.8/logging/__init__.py", line 1085, in emit
    msg = self.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 929, in format
    return fmt.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 668, in format
    record.message = record.getMessage()
  File "/usr/lib/python3.8/logging/__init__.py", line 373, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "/usr/lib/python3/dist-packages/eventlet/greenthread.py", line 221, in main
    result = function(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/ryu/lib/hub.py", line 59, in _launch
    return func(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/ryu/base/app_manager.py", line 290, in _event_loop
    handler(ev)
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 391, in _flow_stats_reply_handler
    self.logger.info('body is ', body)
Message: 'body is '
Arguments: ([OFPFlowStats(byte_count=1808400,cookie=0,duration_nsec=286000000,duration_sec=6,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65535,port=4294967293,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_dst': '01:80:c2:00:00:0e', 'eth_type': 35020}),packet_count=30140,priority=65535,table_id=0), OFPFlowStats(byte_count=1355732,cookie=0,duration_nsec=305000000,duration_sec=6,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.1', 'ip_proto': 1}),packet_count=13834,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=305000000,duration_sec=6,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=3,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.3', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=1355732,cookie=0,duration_nsec=305000000,duration_sec=6,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=1,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.2', 'ip_proto': 1}),packet_count=13834,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=305000000,duration_sec=6,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=3,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.4', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=264558,cookie=0,duration_nsec=114000000,duration_sec=641,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=104,match=OFPMatch(oxm_fields={'in_port': 1, 'eth_src': '00:00:00:00:00:02', 'eth_dst': '00:00:00:00:00:01'}),packet_count=6299,priority=1,table_id=0), OFPFlowStats(byte_count=1260,cookie=0,duration_nsec=959000000,duration_sec=635,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=1,type=0)],len=24,type=4)],length=104,match=OFPMatch(oxm_fields={'in_port': 2, 'eth_src': '00:00:00:00:00:01', 'eth_dst': '00:00:00:00:00:02'}),packet_count=30,priority=1,table_id=0), OFPFlowStats(byte_count=287826,cookie=0,duration_nsec=308000000,duration_sec=6,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65535,port=4294967293,type=0)],len=24,type=4)],length=80,match=OFPMatch(oxm_fields={}),packet_count=6743,priority=0,table_id=0)],)
datapath         in-port  eth-dst           out-port packets  bytes
---------------- -------- ----------------- -------- -------- --------
Switch: Exception occurred during handler processing. Backtrace from offending handler [_flow_stats_reply_handler] servicing event [EventOFPFlowStatsReply] follows.
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/ryu/base/app_manager.py", line 290, in _event_loop
    handler(ev)
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 398, in _flow_stats_reply_handler
    for stat in sorted([flow for flow in body if flow.priority == 10],
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 399, in <lambda>
    key=lambda flow: (flow.match['in_port'],
  File "/usr/lib/python3/dist-packages/ryu/ofproto/ofproto_v1_3_parser.py", line 904, in __getitem__
    return dict(self._fields2)[key]
KeyError: 'in_port'
datapath         port     rx-pkts  rx-bytes rx-error tx-pkts  tx-bytes tx-error
---------------- -------- -------- -------- -------- -------- -------- --------
0000000000000002        1    20549  1638598        0    35638  2552617        0
0000000000000002        2    32115  2397561        0    38382  2660775        0
0000000000000002        3    18246  1040267        0    18248  1040387        0
0000000000000002 fffffffe        0        0        0        0        0        0
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.8/logging/__init__.py", line 1085, in emit
    msg = self.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 929, in format
    return fmt.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 668, in format
    record.message = record.getMessage()
  File "/usr/lib/python3.8/logging/__init__.py", line 373, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "/usr/lib/python3/dist-packages/eventlet/greenthread.py", line 221, in main
    result = function(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/ryu/lib/hub.py", line 59, in _launch
    return func(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/ryu/base/app_manager.py", line 290, in _event_loop
    handler(ev)
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 391, in _flow_stats_reply_handler
    self.logger.info('body is ', body)
Message: 'body is '
Arguments: ([OFPFlowStats(byte_count=1808580,cookie=0,duration_nsec=292000000,duration_sec=6,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65535,port=4294967293,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_dst': '01:80:c2:00:00:0e', 'eth_type': 35020}),packet_count=30143,priority=65535,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=334000000,duration_sec=6,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=3,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.1', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=334000000,duration_sec=6,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.3', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=334000000,duration_sec=6,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.2', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=334000000,duration_sec=6,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=1,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.4', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=270506,cookie=0,duration_nsec=337000000,duration_sec=6,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65535,port=4294967293,type=0)],len=24,type=4)],length=80,match=OFPMatch(oxm_fields={}),packet_count=6346,priority=0,table_id=0)],)
datapath         in-port  eth-dst           out-port packets  bytes
---------------- -------- ----------------- -------- -------- --------
Switch: Exception occurred during handler processing. Backtrace from offending handler [_flow_stats_reply_handler] servicing event [EventOFPFlowStatsReply] follows.
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/ryu/base/app_manager.py", line 290, in _event_loop
    handler(ev)
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 398, in _flow_stats_reply_handler
    for stat in sorted([flow for flow in body if flow.priority == 10],
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 399, in <lambda>
    key=lambda flow: (flow.match['in_port'],
  File "/usr/lib/python3/dist-packages/ryu/ofproto/ofproto_v1_3_parser.py", line 904, in __getitem__
    return dict(self._fields2)[key]
KeyError: 'in_port'
datapath         port     rx-pkts  rx-bytes rx-error tx-pkts  tx-bytes tx-error
---------------- -------- -------- -------- -------- -------- -------- --------
0000000000000004        1       22     1732        0    21384  1172177        0
0000000000000004        2    18249  1040457        0    18249  1040457        0
0000000000000004        3    18249  1040447        0    18248  1040387        0
0000000000000004 fffffffe        0        0        0        0        0        0
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.8/logging/__init__.py", line 1085, in emit
    msg = self.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 929, in format
    return fmt.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 668, in format
    record.message = record.getMessage()
  File "/usr/lib/python3.8/logging/__init__.py", line 373, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "/usr/lib/python3/dist-packages/eventlet/greenthread.py", line 221, in main
    result = function(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/ryu/lib/hub.py", line 59, in _launch
    return func(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/ryu/base/app_manager.py", line 290, in _event_loop
    handler(ev)
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 391, in _flow_stats_reply_handler
    self.logger.info('body is ', body)
Message: 'body is '
Arguments: ([OFPFlowStats(byte_count=1808520,cookie=0,duration_nsec=288000000,duration_sec=6,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65535,port=4294967293,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_dst': '01:80:c2:00:00:0e', 'eth_type': 35020}),packet_count=30142,priority=65535,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=312000000,duration_sec=6,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.1', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=312000000,duration_sec=6,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=1,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.3', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=312000000,duration_sec=6,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.2', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=312000000,duration_sec=6,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=3,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.4', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=270660,cookie=0,duration_nsec=312000000,duration_sec=6,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65535,port=4294967293,type=0)],len=24,type=4)],length=80,match=OFPMatch(oxm_fields={}),packet_count=6346,priority=0,table_id=0)],)
datapath         in-port  eth-dst           out-port packets  bytes
---------------- -------- ----------------- -------- -------- --------
Switch: Exception occurred during handler processing. Backtrace from offending handler [_flow_stats_reply_handler] servicing event [EventOFPFlowStatsReply] follows.
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/ryu/base/app_manager.py", line 290, in _event_loop
    handler(ev)
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 398, in _flow_stats_reply_handler
    for stat in sorted([flow for flow in body if flow.priority == 10],
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 399, in <lambda>
    key=lambda flow: (flow.match['in_port'],
  File "/usr/lib/python3/dist-packages/ryu/ofproto/ofproto_v1_3_parser.py", line 904, in __getitem__
    return dict(self._fields2)[key]
KeyError: 'in_port'
datapath         port     rx-pkts  rx-bytes rx-error tx-pkts  tx-bytes tx-error
---------------- -------- -------- -------- -------- -------- -------- --------
0000000000000003        1       21     1662        0    21381  1171987        0
0000000000000003        2    18248  1040387        0    18246  1040267        0
0000000000000003        3    18249  1040457        0    18249  1040457        0
0000000000000003 fffffffe        0        0        0        0        0        0
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.8/logging/__init__.py", line 1085, in emit
    msg = self.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 929, in format
    return fmt.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 668, in format
    record.message = record.getMessage()
  File "/usr/lib/python3.8/logging/__init__.py", line 373, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "/usr/lib/python3/dist-packages/eventlet/greenthread.py", line 221, in main
    result = function(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/ryu/lib/hub.py", line 59, in _launch
    return func(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/ryu/base/app_manager.py", line 290, in _event_loop
    handler(ev)
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 391, in _flow_stats_reply_handler
    self.logger.info('body is ', body)
Message: 'body is '
Arguments: ([OFPFlowStats(byte_count=1808640,cookie=0,duration_nsec=280000000,duration_sec=7,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65535,port=4294967293,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_dst': '01:80:c2:00:00:0e', 'eth_type': 35020}),packet_count=30144,priority=65535,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=304000000,duration_sec=7,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.1', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=304000000,duration_sec=7,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=1,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.3', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=304000000,duration_sec=7,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.2', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=304000000,duration_sec=7,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=3,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.4', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=270660,cookie=0,duration_nsec=304000000,duration_sec=7,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65535,port=4294967293,type=0)],len=24,type=4)],length=80,match=OFPMatch(oxm_fields={}),packet_count=6346,priority=0,table_id=0)],)
datapath         in-port  eth-dst           out-port packets  bytes
---------------- -------- ----------------- -------- -------- --------
Switch: Exception occurred during handler processing. Backtrace from offending handler [_flow_stats_reply_handler] servicing event [EventOFPFlowStatsReply] follows.
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/ryu/base/app_manager.py", line 290, in _event_loop
    handler(ev)
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 398, in _flow_stats_reply_handler
    for stat in sorted([flow for flow in body if flow.priority == 10],
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 399, in <lambda>
    key=lambda flow: (flow.match['in_port'],
  File "/usr/lib/python3/dist-packages/ryu/ofproto/ofproto_v1_3_parser.py", line 904, in __getitem__
    return dict(self._fields2)[key]
KeyError: 'in_port'
datapath         port     rx-pkts  rx-bytes rx-error tx-pkts  tx-bytes tx-error
---------------- -------- -------- -------- -------- -------- -------- --------
0000000000000003        1       21     1662        0    21382  1172047        0
0000000000000003        2    18249  1040447        0    18247  1040327        0
0000000000000003        3    18250  1040517        0    18250  1040517        0
0000000000000003 fffffffe        0        0        0        0        0        0
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.8/logging/__init__.py", line 1085, in emit
    msg = self.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 929, in format
    return fmt.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 668, in format
    record.message = record.getMessage()
  File "/usr/lib/python3.8/logging/__init__.py", line 373, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "/usr/lib/python3/dist-packages/eventlet/greenthread.py", line 221, in main
    result = function(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/ryu/lib/hub.py", line 59, in _launch
    return func(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/ryu/base/app_manager.py", line 290, in _event_loop
    handler(ev)
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 391, in _flow_stats_reply_handler
    self.logger.info('body is ', body)
Message: 'body is '
Arguments: ([OFPFlowStats(byte_count=1808700,cookie=0,duration_nsec=295000000,duration_sec=7,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65535,port=4294967293,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_dst': '01:80:c2:00:00:0e', 'eth_type': 35020}),packet_count=30145,priority=65535,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=337000000,duration_sec=7,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=3,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.1', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=337000000,duration_sec=7,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.3', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=337000000,duration_sec=7,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.2', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=337000000,duration_sec=7,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=1,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.4', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=270506,cookie=0,duration_nsec=340000000,duration_sec=7,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65535,port=4294967293,type=0)],len=24,type=4)],length=80,match=OFPMatch(oxm_fields={}),packet_count=6346,priority=0,table_id=0)],)
datapath         in-port  eth-dst           out-port packets  bytes
---------------- -------- ----------------- -------- -------- --------
Switch: Exception occurred during handler processing. Backtrace from offending handler [_flow_stats_reply_handler] servicing event [EventOFPFlowStatsReply] follows.
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/ryu/base/app_manager.py", line 290, in _event_loop
    handler(ev)
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 398, in _flow_stats_reply_handler
    for stat in sorted([flow for flow in body if flow.priority == 10],
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 399, in <lambda>
    key=lambda flow: (flow.match['in_port'],
  File "/usr/lib/python3/dist-packages/ryu/ofproto/ofproto_v1_3_parser.py", line 904, in __getitem__
    return dict(self._fields2)[key]
KeyError: 'in_port'
datapath         port     rx-pkts  rx-bytes rx-error tx-pkts  tx-bytes tx-error
---------------- -------- -------- -------- -------- -------- -------- --------
0000000000000004        1       22     1732        0    21385  1172237        0
0000000000000004        2    18250  1040517        0    18250  1040517        0
0000000000000004        3    18250  1040507        0    18249  1040447        0
0000000000000004 fffffffe        0        0        0        0        0        0
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.8/logging/__init__.py", line 1085, in emit
    msg = self.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 929, in format
    return fmt.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 668, in format
    record.message = record.getMessage()
  File "/usr/lib/python3.8/logging/__init__.py", line 373, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "/usr/lib/python3/dist-packages/eventlet/greenthread.py", line 221, in main
    result = function(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/ryu/lib/hub.py", line 59, in _launch
    return func(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/ryu/base/app_manager.py", line 290, in _event_loop
    handler(ev)
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 391, in _flow_stats_reply_handler
    self.logger.info('body is ', body)
Message: 'body is '
Arguments: ([OFPFlowStats(byte_count=1808700,cookie=0,duration_nsec=280000000,duration_sec=7,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65535,port=4294967293,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_dst': '01:80:c2:00:00:0e', 'eth_type': 35020}),packet_count=30145,priority=65535,table_id=0), OFPFlowStats(byte_count=1355830,cookie=0,duration_nsec=313000000,duration_sec=7,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=1,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.1', 'ip_proto': 1}),packet_count=13835,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=313000000,duration_sec=7,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.3', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=1355830,cookie=0,duration_nsec=313000000,duration_sec=7,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.2', 'ip_proto': 1}),packet_count=13835,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=313000000,duration_sec=7,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=3,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.4', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=264558,cookie=0,duration_nsec=90000000,duration_sec=642,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=1,type=0)],len=24,type=4)],length=104,match=OFPMatch(oxm_fields={'in_port': 2, 'eth_src': '00:00:00:00:00:02', 'eth_dst': '00:00:00:00:00:01'}),packet_count=6299,priority=1,table_id=0), OFPFlowStats(byte_count=1260,cookie=0,duration_nsec=972000000,duration_sec=636,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=104,match=OFPMatch(oxm_fields={'in_port': 1, 'eth_src': '00:00:00:00:00:01', 'eth_dst': '00:00:00:00:00:02'}),packet_count=30,priority=1,table_id=0), OFPFlowStats(byte_count=290586,cookie=0,duration_nsec=326000000,duration_sec=7,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65535,port=4294967293,type=0)],len=24,type=4)],length=80,match=OFPMatch(oxm_fields={}),packet_count=6790,priority=0,table_id=0)],)
datapath         in-port  eth-dst           out-port packets  bytes
---------------- -------- ----------------- -------- -------- --------
Switch: Exception occurred during handler processing. Backtrace from offending handler [_flow_stats_reply_handler] servicing event [EventOFPFlowStatsReply] follows.
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/ryu/base/app_manager.py", line 290, in _event_loop
    handler(ev)
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 398, in _flow_stats_reply_handler
    for stat in sorted([flow for flow in body if flow.priority == 10],
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 399, in <lambda>
    key=lambda flow: (flow.match['in_port'],
  File "/usr/lib/python3/dist-packages/ryu/ofproto/ofproto_v1_3_parser.py", line 904, in __getitem__
    return dict(self._fields2)[key]
KeyError: 'in_port'
datapath         port     rx-pkts  rx-bytes rx-error tx-pkts  tx-bytes tx-error
---------------- -------- -------- -------- -------- -------- -------- --------
0000000000000001        1    14316  1376720        0    41909  2816025        0
0000000000000001        2    38383  2660835        0    32116  2397621        0
0000000000000001        3    18249  1040447        0    18250  1040507        0
0000000000000001 fffffffe        0        0        0        0        0        0
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.8/logging/__init__.py", line 1085, in emit
    msg = self.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 929, in format
    return fmt.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 668, in format
    record.message = record.getMessage()
  File "/usr/lib/python3.8/logging/__init__.py", line 373, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "/usr/lib/python3/dist-packages/eventlet/greenthread.py", line 221, in main
    result = function(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/ryu/lib/hub.py", line 59, in _launch
    return func(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/ryu/base/app_manager.py", line 290, in _event_loop
    handler(ev)
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 391, in _flow_stats_reply_handler
    self.logger.info('body is ', body)
Message: 'body is '
Arguments: ([OFPFlowStats(byte_count=1808520,cookie=0,duration_nsec=286000000,duration_sec=7,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65535,port=4294967293,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_dst': '01:80:c2:00:00:0e', 'eth_type': 35020}),packet_count=30142,priority=65535,table_id=0), OFPFlowStats(byte_count=1355830,cookie=0,duration_nsec=305000000,duration_sec=7,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.1', 'ip_proto': 1}),packet_count=13835,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=305000000,duration_sec=7,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=3,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.3', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=1355830,cookie=0,duration_nsec=305000000,duration_sec=7,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=1,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.2', 'ip_proto': 1}),packet_count=13835,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=305000000,duration_sec=7,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=3,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.4', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=264558,cookie=0,duration_nsec=114000000,duration_sec=642,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=104,match=OFPMatch(oxm_fields={'in_port': 1, 'eth_src': '00:00:00:00:00:02', 'eth_dst': '00:00:00:00:00:01'}),packet_count=6299,priority=1,table_id=0), OFPFlowStats(byte_count=1260,cookie=0,duration_nsec=959000000,duration_sec=636,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=1,type=0)],len=24,type=4)],length=104,match=OFPMatch(oxm_fields={'in_port': 2, 'eth_src': '00:00:00:00:00:01', 'eth_dst': '00:00:00:00:00:02'}),packet_count=30,priority=1,table_id=0), OFPFlowStats(byte_count=287826,cookie=0,duration_nsec=308000000,duration_sec=7,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65535,port=4294967293,type=0)],len=24,type=4)],length=80,match=OFPMatch(oxm_fields={}),packet_count=6743,priority=0,table_id=0)],)
datapath         in-port  eth-dst           out-port packets  bytes
---------------- -------- ----------------- -------- -------- --------
Switch: Exception occurred during handler processing. Backtrace from offending handler [_flow_stats_reply_handler] servicing event [EventOFPFlowStatsReply] follows.
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/ryu/base/app_manager.py", line 290, in _event_loop
    handler(ev)
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 398, in _flow_stats_reply_handler
    for stat in sorted([flow for flow in body if flow.priority == 10],
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 399, in <lambda>
    key=lambda flow: (flow.match['in_port'],
  File "/usr/lib/python3/dist-packages/ryu/ofproto/ofproto_v1_3_parser.py", line 904, in __getitem__
    return dict(self._fields2)[key]
KeyError: 'in_port'
datapath         port     rx-pkts  rx-bytes rx-error tx-pkts  tx-bytes tx-error
---------------- -------- -------- -------- -------- -------- -------- --------
0000000000000002        1    20549  1638598        0    35639  2552677        0
0000000000000002        2    32116  2397621        0    38383  2660835        0
0000000000000002        3    18247  1040327        0    18249  1040447        0
0000000000000002 fffffffe        0        0        0        0        0        0
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.8/logging/__init__.py", line 1085, in emit
    msg = self.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 929, in format
    return fmt.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 668, in format
    record.message = record.getMessage()
  File "/usr/lib/python3.8/logging/__init__.py", line 373, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "/usr/lib/python3/dist-packages/eventlet/greenthread.py", line 221, in main
    result = function(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/ryu/lib/hub.py", line 59, in _launch
    return func(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/ryu/base/app_manager.py", line 290, in _event_loop
    handler(ev)
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 391, in _flow_stats_reply_handler
    self.logger.info('body is ', body)
Message: 'body is '
Arguments: ([OFPFlowStats(byte_count=1808820,cookie=0,duration_nsec=282000000,duration_sec=8,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65535,port=4294967293,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_dst': '01:80:c2:00:00:0e', 'eth_type': 35020}),packet_count=30147,priority=65535,table_id=0), OFPFlowStats(byte_count=1355928,cookie=0,duration_nsec=315000000,duration_sec=8,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=1,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.1', 'ip_proto': 1}),packet_count=13836,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=315000000,duration_sec=8,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.3', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=1355928,cookie=0,duration_nsec=315000000,duration_sec=8,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.2', 'ip_proto': 1}),packet_count=13836,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=315000000,duration_sec=8,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=3,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.4', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=264558,cookie=0,duration_nsec=92000000,duration_sec=643,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=1,type=0)],len=24,type=4)],length=104,match=OFPMatch(oxm_fields={'in_port': 2, 'eth_src': '00:00:00:00:00:02', 'eth_dst': '00:00:00:00:00:01'}),packet_count=6299,priority=1,table_id=0), OFPFlowStats(byte_count=1260,cookie=0,duration_nsec=974000000,duration_sec=637,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=104,match=OFPMatch(oxm_fields={'in_port': 1, 'eth_src': '00:00:00:00:00:01', 'eth_dst': '00:00:00:00:00:02'}),packet_count=30,priority=1,table_id=0), OFPFlowStats(byte_count=290586,cookie=0,duration_nsec=328000000,duration_sec=8,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65535,port=4294967293,type=0)],len=24,type=4)],length=80,match=OFPMatch(oxm_fields={}),packet_count=6790,priority=0,table_id=0)],)
datapath         in-port  eth-dst           out-port packets  bytes
---------------- -------- ----------------- -------- -------- --------
Switch: Exception occurred during handler processing. Backtrace from offending handler [_flow_stats_reply_handler] servicing event [EventOFPFlowStatsReply] follows.
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/ryu/base/app_manager.py", line 290, in _event_loop
    handler(ev)
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 398, in _flow_stats_reply_handler
    for stat in sorted([flow for flow in body if flow.priority == 10],
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 399, in <lambda>
    key=lambda flow: (flow.match['in_port'],
  File "/usr/lib/python3/dist-packages/ryu/ofproto/ofproto_v1_3_parser.py", line 904, in __getitem__
    return dict(self._fields2)[key]
KeyError: 'in_port'
datapath         port     rx-pkts  rx-bytes rx-error tx-pkts  tx-bytes tx-error
---------------- -------- -------- -------- -------- -------- -------- --------
0000000000000001        1    14317  1376818        0    41911  2816183        0
0000000000000001        2    38385  2660993        0    32118  2397779        0
0000000000000001        3    18250  1040507        0    18251  1040567        0
0000000000000001 fffffffe        0        0        0        0        0        0
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.8/logging/__init__.py", line 1085, in emit
    msg = self.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 929, in format
    return fmt.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 668, in format
    record.message = record.getMessage()
  File "/usr/lib/python3.8/logging/__init__.py", line 373, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "/usr/lib/python3/dist-packages/eventlet/greenthread.py", line 221, in main
    result = function(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/ryu/lib/hub.py", line 59, in _launch
    return func(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/ryu/base/app_manager.py", line 290, in _event_loop
    handler(ev)
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 391, in _flow_stats_reply_handler
    self.logger.info('body is ', body)
Message: 'body is '
Arguments: ([OFPFlowStats(byte_count=1808700,cookie=0,duration_nsec=286000000,duration_sec=8,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65535,port=4294967293,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_dst': '01:80:c2:00:00:0e', 'eth_type': 35020}),packet_count=30145,priority=65535,table_id=0), OFPFlowStats(byte_count=1355928,cookie=0,duration_nsec=305000000,duration_sec=8,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.1', 'ip_proto': 1}),packet_count=13836,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=305000000,duration_sec=8,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=3,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.3', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=1355928,cookie=0,duration_nsec=305000000,duration_sec=8,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=1,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.2', 'ip_proto': 1}),packet_count=13836,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=305000000,duration_sec=8,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=3,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.4', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=264558,cookie=0,duration_nsec=114000000,duration_sec=643,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=104,match=OFPMatch(oxm_fields={'in_port': 1, 'eth_src': '00:00:00:00:00:02', 'eth_dst': '00:00:00:00:00:01'}),packet_count=6299,priority=1,table_id=0), OFPFlowStats(byte_count=1260,cookie=0,duration_nsec=959000000,duration_sec=637,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=1,type=0)],len=24,type=4)],length=104,match=OFPMatch(oxm_fields={'in_port': 2, 'eth_src': '00:00:00:00:00:01', 'eth_dst': '00:00:00:00:00:02'}),packet_count=30,priority=1,table_id=0), OFPFlowStats(byte_count=287826,cookie=0,duration_nsec=308000000,duration_sec=8,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65535,port=4294967293,type=0)],len=24,type=4)],length=80,match=OFPMatch(oxm_fields={}),packet_count=6743,priority=0,table_id=0)],)
datapath         in-port  eth-dst           out-port packets  bytes
---------------- -------- ----------------- -------- -------- --------
Switch: Exception occurred during handler processing. Backtrace from offending handler [_flow_stats_reply_handler] servicing event [EventOFPFlowStatsReply] follows.
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/ryu/base/app_manager.py", line 290, in _event_loop
    handler(ev)
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 398, in _flow_stats_reply_handler
    for stat in sorted([flow for flow in body if flow.priority == 10],
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 399, in <lambda>
    key=lambda flow: (flow.match['in_port'],
  File "/usr/lib/python3/dist-packages/ryu/ofproto/ofproto_v1_3_parser.py", line 904, in __getitem__
    return dict(self._fields2)[key]
KeyError: 'in_port'
datapath         port     rx-pkts  rx-bytes rx-error tx-pkts  tx-bytes tx-error
---------------- -------- -------- -------- -------- -------- -------- --------
0000000000000002        1    20550  1638696        0    35641  2552835        0
0000000000000002        2    32118  2397779        0    38385  2660993        0
0000000000000002        3    18248  1040387        0    18250  1040507        0
0000000000000002 fffffffe        0        0        0        0        0        0
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.8/logging/__init__.py", line 1085, in emit
    msg = self.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 929, in format
    return fmt.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 668, in format
    record.message = record.getMessage()
  File "/usr/lib/python3.8/logging/__init__.py", line 373, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "/usr/lib/python3/dist-packages/eventlet/greenthread.py", line 221, in main
    result = function(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/ryu/lib/hub.py", line 59, in _launch
    return func(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/ryu/base/app_manager.py", line 290, in _event_loop
    handler(ev)
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 391, in _flow_stats_reply_handler
    self.logger.info('body is ', body)
Message: 'body is '
Arguments: ([OFPFlowStats(byte_count=1808820,cookie=0,duration_nsec=296000000,duration_sec=8,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65535,port=4294967293,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_dst': '01:80:c2:00:00:0e', 'eth_type': 35020}),packet_count=30147,priority=65535,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=338000000,duration_sec=8,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=3,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.1', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=338000000,duration_sec=8,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.3', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=338000000,duration_sec=8,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.2', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=338000000,duration_sec=8,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=1,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.4', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=270506,cookie=0,duration_nsec=341000000,duration_sec=8,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65535,port=4294967293,type=0)],len=24,type=4)],length=80,match=OFPMatch(oxm_fields={}),packet_count=6346,priority=0,table_id=0)],)
datapath         in-port  eth-dst           out-port packets  bytes
---------------- -------- ----------------- -------- -------- --------
Switch: Exception occurred during handler processing. Backtrace from offending handler [_flow_stats_reply_handler] servicing event [EventOFPFlowStatsReply] follows.
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/ryu/base/app_manager.py", line 290, in _event_loop
    handler(ev)
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 398, in _flow_stats_reply_handler
    for stat in sorted([flow for flow in body if flow.priority == 10],
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 399, in <lambda>
    key=lambda flow: (flow.match['in_port'],
  File "/usr/lib/python3/dist-packages/ryu/ofproto/ofproto_v1_3_parser.py", line 904, in __getitem__
    return dict(self._fields2)[key]
KeyError: 'in_port'
datapath         port     rx-pkts  rx-bytes rx-error tx-pkts  tx-bytes tx-error
---------------- -------- -------- -------- -------- -------- -------- --------
0000000000000004        1       22     1732        0    21386  1172297        0
0000000000000004        2    18251  1040577        0    18251  1040577        0
0000000000000004        3    18251  1040567        0    18250  1040507        0
0000000000000004 fffffffe        0        0        0        0        0        0
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.8/logging/__init__.py", line 1085, in emit
    msg = self.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 929, in format
    return fmt.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 668, in format
    record.message = record.getMessage()
  File "/usr/lib/python3.8/logging/__init__.py", line 373, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "/usr/lib/python3/dist-packages/eventlet/greenthread.py", line 221, in main
    result = function(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/ryu/lib/hub.py", line 59, in _launch
    return func(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/ryu/base/app_manager.py", line 290, in _event_loop
    handler(ev)
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 391, in _flow_stats_reply_handler
    self.logger.info('body is ', body)
Message: 'body is '
Arguments: ([OFPFlowStats(byte_count=1808760,cookie=0,duration_nsec=287000000,duration_sec=8,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65535,port=4294967293,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_dst': '01:80:c2:00:00:0e', 'eth_type': 35020}),packet_count=30146,priority=65535,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=311000000,duration_sec=8,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.1', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=311000000,duration_sec=8,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=1,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.3', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=311000000,duration_sec=8,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.2', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=311000000,duration_sec=8,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=3,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.4', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=270660,cookie=0,duration_nsec=311000000,duration_sec=8,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65535,port=4294967293,type=0)],len=24,type=4)],length=80,match=OFPMatch(oxm_fields={}),packet_count=6346,priority=0,table_id=0)],)
datapath         in-port  eth-dst           out-port packets  bytes
---------------- -------- ----------------- -------- -------- --------
Switch: Exception occurred during handler processing. Backtrace from offending handler [_flow_stats_reply_handler] servicing event [EventOFPFlowStatsReply] follows.
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/ryu/base/app_manager.py", line 290, in _event_loop
    handler(ev)
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 398, in _flow_stats_reply_handler
    for stat in sorted([flow for flow in body if flow.priority == 10],
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 399, in <lambda>
    key=lambda flow: (flow.match['in_port'],
  File "/usr/lib/python3/dist-packages/ryu/ofproto/ofproto_v1_3_parser.py", line 904, in __getitem__
    return dict(self._fields2)[key]
KeyError: 'in_port'
datapath         port     rx-pkts  rx-bytes rx-error tx-pkts  tx-bytes tx-error
---------------- -------- -------- -------- -------- -------- -------- --------
0000000000000003        1       21     1662        0    21383  1172107        0
0000000000000003        2    18250  1040507        0    18248  1040387        0
0000000000000003        3    18251  1040577        0    18251  1040577        0
0000000000000003 fffffffe        0        0        0        0        0        0
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.8/logging/__init__.py", line 1085, in emit
    msg = self.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 929, in format
    return fmt.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 668, in format
    record.message = record.getMessage()
  File "/usr/lib/python3.8/logging/__init__.py", line 373, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "/usr/lib/python3/dist-packages/eventlet/greenthread.py", line 221, in main
    result = function(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/ryu/lib/hub.py", line 59, in _launch
    return func(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/ryu/base/app_manager.py", line 290, in _event_loop
    handler(ev)
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 391, in _flow_stats_reply_handler
    self.logger.info('body is ', body)
Message: 'body is '
Arguments: ([OFPFlowStats(byte_count=1808820,cookie=0,duration_nsec=288000000,duration_sec=9,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65535,port=4294967293,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_dst': '01:80:c2:00:00:0e', 'eth_type': 35020}),packet_count=30147,priority=65535,table_id=0), OFPFlowStats(byte_count=1356026,cookie=0,duration_nsec=307000000,duration_sec=9,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.1', 'ip_proto': 1}),packet_count=13837,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=307000000,duration_sec=9,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=3,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.3', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=1356026,cookie=0,duration_nsec=307000000,duration_sec=9,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=1,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.2', 'ip_proto': 1}),packet_count=13837,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=307000000,duration_sec=9,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=3,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.4', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=264558,cookie=0,duration_nsec=116000000,duration_sec=644,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=104,match=OFPMatch(oxm_fields={'in_port': 1, 'eth_src': '00:00:00:00:00:02', 'eth_dst': '00:00:00:00:00:01'}),packet_count=6299,priority=1,table_id=0), OFPFlowStats(byte_count=1260,cookie=0,duration_nsec=961000000,duration_sec=638,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=1,type=0)],len=24,type=4)],length=104,match=OFPMatch(oxm_fields={'in_port': 2, 'eth_src': '00:00:00:00:00:01', 'eth_dst': '00:00:00:00:00:02'}),packet_count=30,priority=1,table_id=0), OFPFlowStats(byte_count=287826,cookie=0,duration_nsec=310000000,duration_sec=9,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65535,port=4294967293,type=0)],len=24,type=4)],length=80,match=OFPMatch(oxm_fields={}),packet_count=6743,priority=0,table_id=0)],)
datapath         in-port  eth-dst           out-port packets  bytes
---------------- -------- ----------------- -------- -------- --------
Switch: Exception occurred during handler processing. Backtrace from offending handler [_flow_stats_reply_handler] servicing event [EventOFPFlowStatsReply] follows.
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/ryu/base/app_manager.py", line 290, in _event_loop
    handler(ev)
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 398, in _flow_stats_reply_handler
    for stat in sorted([flow for flow in body if flow.priority == 10],
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 399, in <lambda>
    key=lambda flow: (flow.match['in_port'],
  File "/usr/lib/python3/dist-packages/ryu/ofproto/ofproto_v1_3_parser.py", line 904, in __getitem__
    return dict(self._fields2)[key]
KeyError: 'in_port'
datapath         port     rx-pkts  rx-bytes rx-error tx-pkts  tx-bytes tx-error
---------------- -------- -------- -------- -------- -------- -------- --------
0000000000000002        1    20551  1638794        0    35643  2552993        0
0000000000000002        2    32120  2397937        0    38387  2661151        0
0000000000000002        3    18249  1040447        0    18251  1040567        0
0000000000000002 fffffffe        0        0        0        0        0        0
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.8/logging/__init__.py", line 1085, in emit
    msg = self.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 929, in format
    return fmt.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 668, in format
    record.message = record.getMessage()
  File "/usr/lib/python3.8/logging/__init__.py", line 373, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "/usr/lib/python3/dist-packages/eventlet/greenthread.py", line 221, in main
    result = function(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/ryu/lib/hub.py", line 59, in _launch
    return func(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/ryu/base/app_manager.py", line 290, in _event_loop
    handler(ev)
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 391, in _flow_stats_reply_handler
    self.logger.info('body is ', body)
Message: 'body is '
Arguments: ([OFPFlowStats(byte_count=1809000,cookie=0,duration_nsec=284000000,duration_sec=9,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65535,port=4294967293,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_dst': '01:80:c2:00:00:0e', 'eth_type': 35020}),packet_count=30150,priority=65535,table_id=0), OFPFlowStats(byte_count=1356026,cookie=0,duration_nsec=317000000,duration_sec=9,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=1,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.1', 'ip_proto': 1}),packet_count=13837,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=317000000,duration_sec=9,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.3', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=1356026,cookie=0,duration_nsec=317000000,duration_sec=9,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.2', 'ip_proto': 1}),packet_count=13837,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=317000000,duration_sec=9,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=3,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.4', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=264558,cookie=0,duration_nsec=94000000,duration_sec=644,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=1,type=0)],len=24,type=4)],length=104,match=OFPMatch(oxm_fields={'in_port': 2, 'eth_src': '00:00:00:00:00:02', 'eth_dst': '00:00:00:00:00:01'}),packet_count=6299,priority=1,table_id=0), OFPFlowStats(byte_count=1260,cookie=0,duration_nsec=976000000,duration_sec=638,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=104,match=OFPMatch(oxm_fields={'in_port': 1, 'eth_src': '00:00:00:00:00:01', 'eth_dst': '00:00:00:00:00:02'}),packet_count=30,priority=1,table_id=0), OFPFlowStats(byte_count=290586,cookie=0,duration_nsec=330000000,duration_sec=9,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65535,port=4294967293,type=0)],len=24,type=4)],length=80,match=OFPMatch(oxm_fields={}),packet_count=6790,priority=0,table_id=0)],)
datapath         in-port  eth-dst           out-port packets  bytes
---------------- -------- ----------------- -------- -------- --------
Switch: Exception occurred during handler processing. Backtrace from offending handler [_flow_stats_reply_handler] servicing event [EventOFPFlowStatsReply] follows.
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/ryu/base/app_manager.py", line 290, in _event_loop
    handler(ev)
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 398, in _flow_stats_reply_handler
    for stat in sorted([flow for flow in body if flow.priority == 10],
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 399, in <lambda>
    key=lambda flow: (flow.match['in_port'],
  File "/usr/lib/python3/dist-packages/ryu/ofproto/ofproto_v1_3_parser.py", line 904, in __getitem__
    return dict(self._fields2)[key]
KeyError: 'in_port'
datapath         port     rx-pkts  rx-bytes rx-error tx-pkts  tx-bytes tx-error
---------------- -------- -------- -------- -------- -------- -------- --------
0000000000000001        1    14318  1376916        0    41913  2816341        0
0000000000000001        2    38387  2661151        0    32120  2397937        0
0000000000000001        3    18251  1040567        0    18252  1040627        0
0000000000000001 fffffffe        0        0        0        0        0        0
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.8/logging/__init__.py", line 1085, in emit
    msg = self.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 929, in format
    return fmt.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 668, in format
    record.message = record.getMessage()
  File "/usr/lib/python3.8/logging/__init__.py", line 373, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "/usr/lib/python3/dist-packages/eventlet/greenthread.py", line 221, in main
    result = function(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/ryu/lib/hub.py", line 59, in _launch
    return func(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/ryu/base/app_manager.py", line 290, in _event_loop
    handler(ev)
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 391, in _flow_stats_reply_handler
    self.logger.info('body is ', body)
Message: 'body is '
Arguments: ([OFPFlowStats(byte_count=1808940,cookie=0,duration_nsec=298000000,duration_sec=9,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65535,port=4294967293,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_dst': '01:80:c2:00:00:0e', 'eth_type': 35020}),packet_count=30149,priority=65535,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=340000000,duration_sec=9,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=3,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.1', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=340000000,duration_sec=9,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.3', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=340000000,duration_sec=9,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.2', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=340000000,duration_sec=9,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=1,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.4', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=270506,cookie=0,duration_nsec=343000000,duration_sec=9,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65535,port=4294967293,type=0)],len=24,type=4)],length=80,match=OFPMatch(oxm_fields={}),packet_count=6346,priority=0,table_id=0)],)
datapath         in-port  eth-dst           out-port packets  bytes
---------------- -------- ----------------- -------- -------- --------
Switch: Exception occurred during handler processing. Backtrace from offending handler [_flow_stats_reply_handler] servicing event [EventOFPFlowStatsReply] follows.
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/ryu/base/app_manager.py", line 290, in _event_loop
    handler(ev)
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 398, in _flow_stats_reply_handler
    for stat in sorted([flow for flow in body if flow.priority == 10],
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 399, in <lambda>
    key=lambda flow: (flow.match['in_port'],
  File "/usr/lib/python3/dist-packages/ryu/ofproto/ofproto_v1_3_parser.py", line 904, in __getitem__
    return dict(self._fields2)[key]
KeyError: 'in_port'
datapath         port     rx-pkts  rx-bytes rx-error tx-pkts  tx-bytes tx-error
---------------- -------- -------- -------- -------- -------- -------- --------
0000000000000004        1       22     1732        0    21387  1172357        0
0000000000000004        2    18252  1040637        0    18252  1040637        0
0000000000000004        3    18252  1040627        0    18251  1040567        0
0000000000000004 fffffffe        0        0        0        0        0        0
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.8/logging/__init__.py", line 1085, in emit
    msg = self.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 929, in format
    return fmt.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 668, in format
    record.message = record.getMessage()
  File "/usr/lib/python3.8/logging/__init__.py", line 373, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "/usr/lib/python3/dist-packages/eventlet/greenthread.py", line 221, in main
    result = function(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/ryu/lib/hub.py", line 59, in _launch
    return func(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/ryu/base/app_manager.py", line 290, in _event_loop
    handler(ev)
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 391, in _flow_stats_reply_handler
    self.logger.info('body is ', body)
Message: 'body is '
Arguments: ([OFPFlowStats(byte_count=1808880,cookie=0,duration_nsec=288000000,duration_sec=9,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65535,port=4294967293,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_dst': '01:80:c2:00:00:0e', 'eth_type': 35020}),packet_count=30148,priority=65535,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=312000000,duration_sec=9,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.1', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=312000000,duration_sec=9,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=1,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.3', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=312000000,duration_sec=9,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.2', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=312000000,duration_sec=9,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=3,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.4', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=270660,cookie=0,duration_nsec=312000000,duration_sec=9,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65535,port=4294967293,type=0)],len=24,type=4)],length=80,match=OFPMatch(oxm_fields={}),packet_count=6346,priority=0,table_id=0)],)
datapath         in-port  eth-dst           out-port packets  bytes
---------------- -------- ----------------- -------- -------- --------
Switch: Exception occurred during handler processing. Backtrace from offending handler [_flow_stats_reply_handler] servicing event [EventOFPFlowStatsReply] follows.
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/ryu/base/app_manager.py", line 290, in _event_loop
    handler(ev)
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 398, in _flow_stats_reply_handler
    for stat in sorted([flow for flow in body if flow.priority == 10],
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 399, in <lambda>
    key=lambda flow: (flow.match['in_port'],
  File "/usr/lib/python3/dist-packages/ryu/ofproto/ofproto_v1_3_parser.py", line 904, in __getitem__
    return dict(self._fields2)[key]
KeyError: 'in_port'
datapath         port     rx-pkts  rx-bytes rx-error tx-pkts  tx-bytes tx-error
---------------- -------- -------- -------- -------- -------- -------- --------
0000000000000003        1       21     1662        0    21384  1172167        0
0000000000000003        2    18251  1040567        0    18249  1040447        0
0000000000000003        3    18252  1040637        0    18252  1040637        0
0000000000000003 fffffffe        0        0        0        0        0        0
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.8/logging/__init__.py", line 1085, in emit
    msg = self.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 929, in format
    return fmt.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 668, in format
    record.message = record.getMessage()
  File "/usr/lib/python3.8/logging/__init__.py", line 373, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "/usr/lib/python3/dist-packages/eventlet/greenthread.py", line 221, in main
    result = function(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/ryu/lib/hub.py", line 59, in _launch
    return func(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/ryu/base/app_manager.py", line 290, in _event_loop
    handler(ev)
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 391, in _flow_stats_reply_handler
    self.logger.info('body is ', body)
Message: 'body is '
Arguments: ([OFPFlowStats(byte_count=1808940,cookie=0,duration_nsec=290000000,duration_sec=10,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65535,port=4294967293,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_dst': '01:80:c2:00:00:0e', 'eth_type': 35020}),packet_count=30149,priority=65535,table_id=0), OFPFlowStats(byte_count=1356124,cookie=0,duration_nsec=309000000,duration_sec=10,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.1', 'ip_proto': 1}),packet_count=13838,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=309000000,duration_sec=10,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=3,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.3', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=1356124,cookie=0,duration_nsec=309000000,duration_sec=10,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=1,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.2', 'ip_proto': 1}),packet_count=13838,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=309000000,duration_sec=10,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=3,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.4', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=264558,cookie=0,duration_nsec=118000000,duration_sec=645,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=104,match=OFPMatch(oxm_fields={'in_port': 1, 'eth_src': '00:00:00:00:00:02', 'eth_dst': '00:00:00:00:00:01'}),packet_count=6299,priority=1,table_id=0), OFPFlowStats(byte_count=1260,cookie=0,duration_nsec=963000000,duration_sec=639,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=1,type=0)],len=24,type=4)],length=104,match=OFPMatch(oxm_fields={'in_port': 2, 'eth_src': '00:00:00:00:00:01', 'eth_dst': '00:00:00:00:00:02'}),packet_count=30,priority=1,table_id=0), OFPFlowStats(byte_count=287826,cookie=0,duration_nsec=312000000,duration_sec=10,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65535,port=4294967293,type=0)],len=24,type=4)],length=80,match=OFPMatch(oxm_fields={}),packet_count=6743,priority=0,table_id=0)],)
datapath         in-port  eth-dst           out-port packets  bytes
---------------- -------- ----------------- -------- -------- --------
Switch: Exception occurred during handler processing. Backtrace from offending handler [_flow_stats_reply_handler] servicing event [EventOFPFlowStatsReply] follows.
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/ryu/base/app_manager.py", line 290, in _event_loop
    handler(ev)
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 398, in _flow_stats_reply_handler
    for stat in sorted([flow for flow in body if flow.priority == 10],
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 399, in <lambda>
    key=lambda flow: (flow.match['in_port'],
  File "/usr/lib/python3/dist-packages/ryu/ofproto/ofproto_v1_3_parser.py", line 904, in __getitem__
    return dict(self._fields2)[key]
KeyError: 'in_port'
datapath         port     rx-pkts  rx-bytes rx-error tx-pkts  tx-bytes tx-error
---------------- -------- -------- -------- -------- -------- -------- --------
0000000000000002        1    20552  1638892        0    35645  2553151        0
0000000000000002        2    32122  2398095        0    38389  2661309        0
0000000000000002        3    18250  1040507        0    18252  1040627        0
0000000000000002 fffffffe        0        0        0        0        0        0
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.8/logging/__init__.py", line 1085, in emit
    msg = self.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 929, in format
    return fmt.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 668, in format
    record.message = record.getMessage()
  File "/usr/lib/python3.8/logging/__init__.py", line 373, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "/usr/lib/python3/dist-packages/eventlet/greenthread.py", line 221, in main
    result = function(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/ryu/lib/hub.py", line 59, in _launch
    return func(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/ryu/base/app_manager.py", line 290, in _event_loop
    handler(ev)
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 391, in _flow_stats_reply_handler
    self.logger.info('body is ', body)
Message: 'body is '
Arguments: ([OFPFlowStats(byte_count=1809060,cookie=0,duration_nsec=299000000,duration_sec=10,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65535,port=4294967293,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_dst': '01:80:c2:00:00:0e', 'eth_type': 35020}),packet_count=30151,priority=65535,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=341000000,duration_sec=10,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=3,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.1', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=341000000,duration_sec=10,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.3', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=341000000,duration_sec=10,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.2', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=341000000,duration_sec=10,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=1,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.4', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=270506,cookie=0,duration_nsec=344000000,duration_sec=10,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65535,port=4294967293,type=0)],len=24,type=4)],length=80,match=OFPMatch(oxm_fields={}),packet_count=6346,priority=0,table_id=0)],)
datapath         in-port  eth-dst           out-port packets  bytes
---------------- -------- ----------------- -------- -------- --------
Switch: Exception occurred during handler processing. Backtrace from offending handler [_flow_stats_reply_handler] servicing event [EventOFPFlowStatsReply] follows.
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/ryu/base/app_manager.py", line 290, in _event_loop
    handler(ev)
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 398, in _flow_stats_reply_handler
    for stat in sorted([flow for flow in body if flow.priority == 10],
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 399, in <lambda>
    key=lambda flow: (flow.match['in_port'],
  File "/usr/lib/python3/dist-packages/ryu/ofproto/ofproto_v1_3_parser.py", line 904, in __getitem__
    return dict(self._fields2)[key]
KeyError: 'in_port'
datapath         port     rx-pkts  rx-bytes rx-error tx-pkts  tx-bytes tx-error
---------------- -------- -------- -------- -------- -------- -------- --------
0000000000000004        1       22     1732        0    21388  1172417        0
0000000000000004        2    18253  1040697        0    18253  1040697        0
0000000000000004        3    18253  1040687        0    18253  1040687        0
0000000000000004 fffffffe        0        0        0        0        0        0
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.8/logging/__init__.py", line 1085, in emit
    msg = self.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 929, in format
    return fmt.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 668, in format
    record.message = record.getMessage()
  File "/usr/lib/python3.8/logging/__init__.py", line 373, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "/usr/lib/python3/dist-packages/eventlet/greenthread.py", line 221, in main
    result = function(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/ryu/lib/hub.py", line 59, in _launch
    return func(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/ryu/base/app_manager.py", line 290, in _event_loop
    handler(ev)
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 391, in _flow_stats_reply_handler
    self.logger.info('body is ', body)
Message: 'body is '
Arguments: ([OFPFlowStats(byte_count=1809120,cookie=0,duration_nsec=286000000,duration_sec=10,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65535,port=4294967293,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_dst': '01:80:c2:00:00:0e', 'eth_type': 35020}),packet_count=30152,priority=65535,table_id=0), OFPFlowStats(byte_count=1356124,cookie=0,duration_nsec=319000000,duration_sec=10,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=1,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.1', 'ip_proto': 1}),packet_count=13838,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=319000000,duration_sec=10,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.3', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=1356124,cookie=0,duration_nsec=319000000,duration_sec=10,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.2', 'ip_proto': 1}),packet_count=13838,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=319000000,duration_sec=10,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=3,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.4', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=264558,cookie=0,duration_nsec=96000000,duration_sec=645,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=1,type=0)],len=24,type=4)],length=104,match=OFPMatch(oxm_fields={'in_port': 2, 'eth_src': '00:00:00:00:00:02', 'eth_dst': '00:00:00:00:00:01'}),packet_count=6299,priority=1,table_id=0), OFPFlowStats(byte_count=1260,cookie=0,duration_nsec=978000000,duration_sec=639,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=104,match=OFPMatch(oxm_fields={'in_port': 1, 'eth_src': '00:00:00:00:00:01', 'eth_dst': '00:00:00:00:00:02'}),packet_count=30,priority=1,table_id=0), OFPFlowStats(byte_count=290586,cookie=0,duration_nsec=332000000,duration_sec=10,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65535,port=4294967293,type=0)],len=24,type=4)],length=80,match=OFPMatch(oxm_fields={}),packet_count=6790,priority=0,table_id=0)],)
datapath         in-port  eth-dst           out-port packets  bytes
---------------- -------- ----------------- -------- -------- --------
Switch: Exception occurred during handler processing. Backtrace from offending handler [_flow_stats_reply_handler] servicing event [EventOFPFlowStatsReply] follows.
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/ryu/base/app_manager.py", line 290, in _event_loop
    handler(ev)
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 398, in _flow_stats_reply_handler
    for stat in sorted([flow for flow in body if flow.priority == 10],
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 399, in <lambda>
    key=lambda flow: (flow.match['in_port'],
  File "/usr/lib/python3/dist-packages/ryu/ofproto/ofproto_v1_3_parser.py", line 904, in __getitem__
    return dict(self._fields2)[key]
KeyError: 'in_port'
datapath         port     rx-pkts  rx-bytes rx-error tx-pkts  tx-bytes tx-error
---------------- -------- -------- -------- -------- -------- -------- --------
0000000000000001        1    14319  1377014        0    41915  2816499        0
0000000000000001        2    38389  2661309        0    32122  2398095        0
0000000000000001        3    18253  1040687        0    18253  1040687        0
0000000000000001 fffffffe        0        0        0        0        0        0
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.8/logging/__init__.py", line 1085, in emit
    msg = self.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 929, in format
    return fmt.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 668, in format
    record.message = record.getMessage()
  File "/usr/lib/python3.8/logging/__init__.py", line 373, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "/usr/lib/python3/dist-packages/eventlet/greenthread.py", line 221, in main
    result = function(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/ryu/lib/hub.py", line 59, in _launch
    return func(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/ryu/base/app_manager.py", line 290, in _event_loop
    handler(ev)
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 391, in _flow_stats_reply_handler
    self.logger.info('body is ', body)
Message: 'body is '
Arguments: ([OFPFlowStats(byte_count=1809000,cookie=0,duration_nsec=287000000,duration_sec=10,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65535,port=4294967293,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_dst': '01:80:c2:00:00:0e', 'eth_type': 35020}),packet_count=30150,priority=65535,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=311000000,duration_sec=10,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.1', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=311000000,duration_sec=10,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=1,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.3', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=311000000,duration_sec=10,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.2', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=311000000,duration_sec=10,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=3,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.4', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=270660,cookie=0,duration_nsec=311000000,duration_sec=10,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65535,port=4294967293,type=0)],len=24,type=4)],length=80,match=OFPMatch(oxm_fields={}),packet_count=6346,priority=0,table_id=0)],)
datapath         in-port  eth-dst           out-port packets  bytes
---------------- -------- ----------------- -------- -------- --------
Switch: Exception occurred during handler processing. Backtrace from offending handler [_flow_stats_reply_handler] servicing event [EventOFPFlowStatsReply] follows.
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/ryu/base/app_manager.py", line 290, in _event_loop
    handler(ev)
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 398, in _flow_stats_reply_handler
    for stat in sorted([flow for flow in body if flow.priority == 10],
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 399, in <lambda>
    key=lambda flow: (flow.match['in_port'],
  File "/usr/lib/python3/dist-packages/ryu/ofproto/ofproto_v1_3_parser.py", line 904, in __getitem__
    return dict(self._fields2)[key]
KeyError: 'in_port'
datapath         port     rx-pkts  rx-bytes rx-error tx-pkts  tx-bytes tx-error
---------------- -------- -------- -------- -------- -------- -------- --------
0000000000000003        1       21     1662        0    21385  1172227        0
0000000000000003        2    18252  1040627        0    18250  1040507        0
0000000000000003        3    18253  1040697        0    18253  1040697        0
0000000000000003 fffffffe        0        0        0        0        0        0
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.8/logging/__init__.py", line 1085, in emit
    msg = self.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 929, in format
    return fmt.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 668, in format
    record.message = record.getMessage()
  File "/usr/lib/python3.8/logging/__init__.py", line 373, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "/usr/lib/python3/dist-packages/eventlet/greenthread.py", line 221, in main
    result = function(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/ryu/lib/hub.py", line 59, in _launch
    return func(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/ryu/base/app_manager.py", line 290, in _event_loop
    handler(ev)
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 391, in _flow_stats_reply_handler
    self.logger.info('body is ', body)
Message: 'body is '
Arguments: ([OFPFlowStats(byte_count=1809240,cookie=0,duration_nsec=289000000,duration_sec=11,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65535,port=4294967293,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_dst': '01:80:c2:00:00:0e', 'eth_type': 35020}),packet_count=30154,priority=65535,table_id=0), OFPFlowStats(byte_count=1356222,cookie=0,duration_nsec=322000000,duration_sec=11,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=1,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.1', 'ip_proto': 1}),packet_count=13839,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=322000000,duration_sec=11,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.3', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=1356222,cookie=0,duration_nsec=322000000,duration_sec=11,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.2', 'ip_proto': 1}),packet_count=13839,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=322000000,duration_sec=11,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=3,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.4', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=264558,cookie=0,duration_nsec=99000000,duration_sec=646,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=1,type=0)],len=24,type=4)],length=104,match=OFPMatch(oxm_fields={'in_port': 2, 'eth_src': '00:00:00:00:00:02', 'eth_dst': '00:00:00:00:00:01'}),packet_count=6299,priority=1,table_id=0), OFPFlowStats(byte_count=1260,cookie=0,duration_nsec=981000000,duration_sec=640,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=104,match=OFPMatch(oxm_fields={'in_port': 1, 'eth_src': '00:00:00:00:00:01', 'eth_dst': '00:00:00:00:00:02'}),packet_count=30,priority=1,table_id=0), OFPFlowStats(byte_count=290586,cookie=0,duration_nsec=335000000,duration_sec=11,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65535,port=4294967293,type=0)],len=24,type=4)],length=80,match=OFPMatch(oxm_fields={}),packet_count=6790,priority=0,table_id=0)],)
datapath         in-port  eth-dst           out-port packets  bytes
---------------- -------- ----------------- -------- -------- --------
Switch: Exception occurred during handler processing. Backtrace from offending handler [_flow_stats_reply_handler] servicing event [EventOFPFlowStatsReply] follows.
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/ryu/base/app_manager.py", line 290, in _event_loop
    handler(ev)
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 398, in _flow_stats_reply_handler
    for stat in sorted([flow for flow in body if flow.priority == 10],
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 399, in <lambda>
    key=lambda flow: (flow.match['in_port'],
  File "/usr/lib/python3/dist-packages/ryu/ofproto/ofproto_v1_3_parser.py", line 904, in __getitem__
    return dict(self._fields2)[key]
KeyError: 'in_port'
datapath         port     rx-pkts  rx-bytes rx-error tx-pkts  tx-bytes tx-error
---------------- -------- -------- -------- -------- -------- -------- --------
0000000000000001        1    14320  1377112        0    41917  2816657        0
0000000000000001        2    38391  2661467        0    32124  2398253        0
0000000000000001        3    18254  1040747        0    18254  1040747        0
0000000000000001 fffffffe        0        0        0        0        0        0
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.8/logging/__init__.py", line 1085, in emit
    msg = self.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 929, in format
    return fmt.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 668, in format
    record.message = record.getMessage()
  File "/usr/lib/python3.8/logging/__init__.py", line 373, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "/usr/lib/python3/dist-packages/eventlet/greenthread.py", line 221, in main
    result = function(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/ryu/lib/hub.py", line 59, in _launch
    return func(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/ryu/base/app_manager.py", line 290, in _event_loop
    handler(ev)
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 391, in _flow_stats_reply_handler
    self.logger.info('body is ', body)
Message: 'body is '
Arguments: ([OFPFlowStats(byte_count=1809060,cookie=0,duration_nsec=292000000,duration_sec=11,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65535,port=4294967293,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_dst': '01:80:c2:00:00:0e', 'eth_type': 35020}),packet_count=30151,priority=65535,table_id=0), OFPFlowStats(byte_count=1356222,cookie=0,duration_nsec=311000000,duration_sec=11,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.1', 'ip_proto': 1}),packet_count=13839,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=311000000,duration_sec=11,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=3,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.3', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=1356222,cookie=0,duration_nsec=311000000,duration_sec=11,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=1,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.2', 'ip_proto': 1}),packet_count=13839,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=311000000,duration_sec=11,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=3,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.4', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=264558,cookie=0,duration_nsec=120000000,duration_sec=646,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=104,match=OFPMatch(oxm_fields={'in_port': 1, 'eth_src': '00:00:00:00:00:02', 'eth_dst': '00:00:00:00:00:01'}),packet_count=6299,priority=1,table_id=0), OFPFlowStats(byte_count=1260,cookie=0,duration_nsec=965000000,duration_sec=640,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=1,type=0)],len=24,type=4)],length=104,match=OFPMatch(oxm_fields={'in_port': 2, 'eth_src': '00:00:00:00:00:01', 'eth_dst': '00:00:00:00:00:02'}),packet_count=30,priority=1,table_id=0), OFPFlowStats(byte_count=287826,cookie=0,duration_nsec=314000000,duration_sec=11,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65535,port=4294967293,type=0)],len=24,type=4)],length=80,match=OFPMatch(oxm_fields={}),packet_count=6743,priority=0,table_id=0)],)
datapath         in-port  eth-dst           out-port packets  bytes
---------------- -------- ----------------- -------- -------- --------
Switch: Exception occurred during handler processing. Backtrace from offending handler [_flow_stats_reply_handler] servicing event [EventOFPFlowStatsReply] follows.
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/ryu/base/app_manager.py", line 290, in _event_loop
    handler(ev)
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 398, in _flow_stats_reply_handler
    for stat in sorted([flow for flow in body if flow.priority == 10],
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 399, in <lambda>
    key=lambda flow: (flow.match['in_port'],
  File "/usr/lib/python3/dist-packages/ryu/ofproto/ofproto_v1_3_parser.py", line 904, in __getitem__
    return dict(self._fields2)[key]
KeyError: 'in_port'
datapath         port     rx-pkts  rx-bytes rx-error tx-pkts  tx-bytes tx-error
---------------- -------- -------- -------- -------- -------- -------- --------
0000000000000002        1    20553  1638990        0    35647  2553309        0
0000000000000002        2    32124  2398253        0    38391  2661467        0
0000000000000002        3    18251  1040567        0    18253  1040687        0
0000000000000002 fffffffe        0        0        0        0        0        0
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.8/logging/__init__.py", line 1085, in emit
    msg = self.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 929, in format
    return fmt.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 668, in format
    record.message = record.getMessage()
  File "/usr/lib/python3.8/logging/__init__.py", line 373, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "/usr/lib/python3/dist-packages/eventlet/greenthread.py", line 221, in main
    result = function(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/ryu/lib/hub.py", line 59, in _launch
    return func(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/ryu/base/app_manager.py", line 290, in _event_loop
    handler(ev)
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 391, in _flow_stats_reply_handler
    self.logger.info('body is ', body)
Message: 'body is '
Arguments: ([OFPFlowStats(byte_count=1809180,cookie=0,duration_nsec=302000000,duration_sec=11,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65535,port=4294967293,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_dst': '01:80:c2:00:00:0e', 'eth_type': 35020}),packet_count=30153,priority=65535,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=344000000,duration_sec=11,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=3,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.1', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=344000000,duration_sec=11,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.3', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=344000000,duration_sec=11,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.2', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=344000000,duration_sec=11,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=1,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.4', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=270506,cookie=0,duration_nsec=347000000,duration_sec=11,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65535,port=4294967293,type=0)],len=24,type=4)],length=80,match=OFPMatch(oxm_fields={}),packet_count=6346,priority=0,table_id=0)],)
datapath         in-port  eth-dst           out-port packets  bytes
---------------- -------- ----------------- -------- -------- --------
Switch: Exception occurred during handler processing. Backtrace from offending handler [_flow_stats_reply_handler] servicing event [EventOFPFlowStatsReply] follows.
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/ryu/base/app_manager.py", line 290, in _event_loop
    handler(ev)
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 398, in _flow_stats_reply_handler
    for stat in sorted([flow for flow in body if flow.priority == 10],
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 399, in <lambda>
    key=lambda flow: (flow.match['in_port'],
  File "/usr/lib/python3/dist-packages/ryu/ofproto/ofproto_v1_3_parser.py", line 904, in __getitem__
    return dict(self._fields2)[key]
KeyError: 'in_port'
datapath         port     rx-pkts  rx-bytes rx-error tx-pkts  tx-bytes tx-error
---------------- -------- -------- -------- -------- -------- -------- --------
0000000000000004        1       22     1732        0    21389  1172477        0
0000000000000004        2    18254  1040757        0    18254  1040757        0
0000000000000004        3    18254  1040747        0    18254  1040747        0
0000000000000004 fffffffe        0        0        0        0        0        0
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.8/logging/__init__.py", line 1085, in emit
    msg = self.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 929, in format
    return fmt.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 668, in format
    record.message = record.getMessage()
  File "/usr/lib/python3.8/logging/__init__.py", line 373, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "/usr/lib/python3/dist-packages/eventlet/greenthread.py", line 221, in main
    result = function(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/ryu/lib/hub.py", line 59, in _launch
    return func(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/ryu/base/app_manager.py", line 290, in _event_loop
    handler(ev)
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 391, in _flow_stats_reply_handler
    self.logger.info('body is ', body)
Message: 'body is '
Arguments: ([OFPFlowStats(byte_count=1809120,cookie=0,duration_nsec=295000000,duration_sec=11,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65535,port=4294967293,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_dst': '01:80:c2:00:00:0e', 'eth_type': 35020}),packet_count=30152,priority=65535,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=319000000,duration_sec=11,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.1', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=319000000,duration_sec=11,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=1,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.3', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=319000000,duration_sec=11,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=2,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.2', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=0,cookie=0,duration_nsec=319000000,duration_sec=11,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65509,port=3,type=0)],len=24,type=4)],length=96,match=OFPMatch(oxm_fields={'eth_type': 2048, 'ipv4_dst': '10.0.0.4', 'ip_proto': 1}),packet_count=0,priority=10,table_id=0), OFPFlowStats(byte_count=270660,cookie=0,duration_nsec=319000000,duration_sec=11,flags=0,hard_timeout=0,idle_timeout=0,instructions=[OFPInstructionActions(actions=[OFPActionOutput(len=16,max_len=65535,port=4294967293,type=0)],len=24,type=4)],length=80,match=OFPMatch(oxm_fields={}),packet_count=6346,priority=0,table_id=0)],)
datapath         in-port  eth-dst           out-port packets  bytes
---------------- -------- ----------------- -------- -------- --------
Switch: Exception occurred during handler processing. Backtrace from offending handler [_flow_stats_reply_handler] servicing event [EventOFPFlowStatsReply] follows.
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/ryu/base/app_manager.py", line 290, in _event_loop
    handler(ev)
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 398, in _flow_stats_reply_handler
    for stat in sorted([flow for flow in body if flow.priority == 10],
  File "/home/ubuntu/sdn/projects/multi-path/app3_multi-path+monitor.py", line 399, in <lambda>
    key=lambda flow: (flow.match['in_port'],
  File "/usr/lib/python3/dist-packages/ryu/ofproto/ofproto_v1_3_parser.py", line 904, in __getitem__
    return dict(self._fields2)[key]
KeyError: 'in_port'
datapath         port     rx-pkts  rx-bytes rx-error tx-pkts  tx-bytes tx-error
---------------- -------- -------- -------- -------- -------- -------- --------
0000000000000003        1       21     1662        0    21386  1172287        0
0000000000000003        2    18253  1040687        0    18251  1040567        0
0000000000000003        3    18254  1040757        0    18254  1040757        0
0000000000000003 fffffffe        0        0        0        0        0        0
